fix(components/modals): skyConfirmTestingService verify custom button configurations

Blackbaud-NickGlyder opened the pull request:
Ensures that `expectOpen` properly handles custom confirm button configurations, including content and ordering.

https://dev.azure.com/blackbaud/Products/_workitems/edit/2900427

blackbaud-sky-build-user commented:
[Storybook preview](https://blackbaud.github.io/skyux-pr-preview/2267/storybook/)

Component Storybooks:

- [ag-grid-storybook](https://blackbaud.github.io/skyux-pr-preview/2267/storybooks/ag-grid-storybook/)

Apps:

- [code-examples](https://blackbaud.github.io/skyux-pr-preview/2267/code-examples/)
- [dep-graph](https://blackbaud.github.io/skyux-pr-preview/2267/dep-graph/)
- [integration](https://blackbaud.github.io/skyux-pr-preview/2267/integration/)
- [playground](https://blackbaud.github.io/skyux-pr-preview/2267/playground/)

Blackbaud-NickGlyder committed:
fix(components/modals): skyConfirmTestingService verify custom button configurations

Ensures that `expectOpen` properly handles custom confirm button configurations, including content

fix 2900427

Blackbaud-SteveBrush committed:
Merge branch 'main' into nick-confirm-test-fix
